File: backend/schemas/candidate.py
from pydantic import BaseModel, Field


class Experience(BaseModel):
    company: str
    role: str
    description: str


class Education(BaseModel):
    institution: str
    degree: str
    description: str


class Project(BaseModel):
    name: str
    description: str
    technologies: list[str] = Field(default_factory=list)


class CandidateListItem(BaseModel):
    id: int
    name: str
    summary: str | None = None

    model_config = {
        "from_attributes": True,
    }


class CandidateProfile(BaseModel):
    name: str
    summary: str
    skills: list[str] = Field(default_factory=list)
    experience: list[Experience] = Field(default_factory=list)
    education: list[Education] = Field(default_factory=list)
    projects: list[Project] = Field(default_factory=list)
    languages: list[str] = Field(default_factory=list)

File: web/src/types/candidate.ts
export interface CandidateListItem {
  id: number;
  name: string;
  summary: string | null;
}

export interface Experience {
  company: string;
  role: string;
  description: string;
}

export interface Education {
  institution: string;
  degree: string;
  description: string;
}

export interface Project {
  name: string;
  description: string;
  technologies: string[];
}

export interface CandidateProfile {
  name: string;
  summary: string;
  skills: string[];
  experience: Experience[];
  education: Education[];
  projects: Project[];
  languages: string[];
}